### PR TITLE
EDU-857 TOC plugin

### PR DESCRIPTION
**Pull request by ahelmberger**
Closes https://educandu.atlassian.net/browse/EDU-857

### CODE REVIEW

File: src/plugins/table-of-contents/table-of-contents-info.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  cloneContent(content) {
+    return cloneDeep(content);
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  redactContent(content, _targetRoomId) {
+    return cloneDeep(content);
+  }
+
+  // eslint-disable-next-line no-unused-vars
```
Comment (adrianaluchian) [inline src/plugins/table-of-contents/table-of-contents-info.js:57]: As much as I dislike comments, I think in these cases it's a good idea to always have the arguments consistently present 👍

File: src/plugins/table-of-contents/table-of-contents-display.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        level: Number.parseInt(element.tagName.replace(/\D/g, ''), 10) || null
+      }))
+      .filter(node => typeof node.level === 'number' && node.level >= minLevel && node.level <= maxLevel);
+
+    setNodes(nodeList);
+  }, [minLevel, maxLevel]);
+
+  if (!nodes.length) {
+    return null;
+  }
```
Comment (adrianaluchian) [inline src/plugins/table-of-contents/table-of-contents-display.js:23]: I guess we could not early return and just let the display component render whatever it found as nodes, even nothing if there's nothing.
The early return made me think that there might be a danger of using `nodes` beyond this point, but there isn't.

**Comment by adrianaluchian** [comment]
Nice and clean implementation 👏 
Just a tiny complication suggestion from me: maybe we could also have the `width` setting here. With that, the user could basically "style" the table of content a bit (an 80% with would basically add some indentation, which might look better with short headline texts).
Also I think we probably have `width` in all other plugins.

File: src/plugins/table-of-contents/table-of-contents-info.js (hunk 1 above)
Comment (ahelmberger) [reply src/plugins/table-of-contents/table-of-contents-info.js:57]: I think so too, also, when adding a header, we will have to re-introduce them anyways again.

**Pull request merged**